app/Observers/EntityObserver.php

```diff
@@ -156,7 +155,32 @@ protected function saveAttributes(Entity $entity): self
             'is_attributes_private'
         );
         $this->attributeService->saveEntity($data, $entity);
-
+        $sourceId = request()->post('copy_source_id');
+        if (request()->has('replace_mentions') && request()->filled('replace_mentions')) {
+            $source = $source ?? Entity::findOrFail($sourceId);
+            $sourceAttributes = [];
+            $entityAttributes = [];
+            foreach ($source->attributes as $attribute) {
+                array_push($sourceAttributes, '{attribute:' . $attribute->id . '}');
+            }
+            foreach ($entity->attributes as $attribute) {
+                array_push($entityAttributes, '{attribute:' . $attribute->id . '}');
+            }
+            $attributes = array_combine($sourceAttributes, $entityAttributes);
```

Comment:
This whole block can be replaced by
```$entry = Str::replace($sourceAttributes, $entityAttributes, $entity->child->entry);```, using Laravel's [Str helper](https://laravel.com/docs/9.x/helpers#method-str-replace)